source/common/upstream/outlier_detection_impl.cc

```diff
@@ -414,6 +494,11 @@ bool DetectorImpl::enforceEjection(envoy::data::cluster::v3::OutlierEjectionType
   case envoy::data::cluster::v3::FAILURE_PERCENTAGE_LOCAL_ORIGIN:
     return runtime_.snapshot().featureEnabled(EnforcingFailurePercentageLocalOriginRuntime,
                                               config_.enforcingFailurePercentageLocalOrigin());
+  case envoy::data::cluster::v3::EXTENSION:
+    return runtime_.snapshot().featureEnabled(
+        std::string(EnforcingExtensionRuntime) + "." +
+            host->outlierDetector().getFailedExtensionMonitorName(),
```

Comment:
Constructing a new string every time? I will prefer to let the ExtMoniter self to provide a `enforceEjection` interface or return a runtime key.

And how about the case where there are multiple extension monitors are failed?

Reply:
It is a good point about multiple extensions. Let me rethink how to do that. Maybe instead of single monitor, I should pass array/vector of failed monitors and then iterate over them to get `enforce` parameter.